aya/src/programs/mod.rs

```diff
@@ -215,11 +218,17 @@ pub enum ProgramError {
 
 /// A [`Program`] file descriptor.
 #[derive(Copy, Clone)]
-pub struct ProgramFd(RawFd);
+pub struct ProgramFd<'f>(BorrowedFd<'f>);
```

Comment:
As explained in another comment for the breaking doctest this change in behavior can cause borrow checker issues if we need to use one program's fd in a different program in the same bpf object.